src/serial.rs

```diff
@@ -254,15 +288,38 @@ impl<W: Write> Serial<W> {
     /// # Example
     ///
     /// ```rust
+    /// # use std::io::{Error, Result};
+    /// # use std::ops::Deref;
+    /// # use vm_superio::trigger::Trigger;
     /// # use vm_superio::Serial;
     /// # use vmm_sys_util::eventfd::EventFd;
-    /// let intr_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
-    /// let mut serial = Serial::new(intr_evt, Vec::new());
     ///
+    /// struct TestEventFd(EventFd);
```

Comment:
Duplicating this for all public functions is a bit of a bummer. What I saw other projects do instead is to provide just one example at the module level with all public functions, and then just point to that example in the function documentation.

Can we also do this here?

Reply:
Good suggestion, I'll try to see how it looks.